src/components/SendIcon/index.tsx

```diff
@@ -15,17 +15,22 @@ interface SendIconProps {
 }
 
 const SendIcon: React.FC<SendIconProps> = ({type, isDisabled}) => {
-  const sendButtonStyle = React.useCallback(() => {
-    if (isDisabled) return COLORS.gray1;
+  const iconBackgroundColorStyle = React.useCallback(() => {
+    if (isDisabled) return COLORS.gray200;
     if (type === 'SIGNED') return COLORS.signed_primary;
     return COLORS.anon_primary;
   }, [isDisabled, type]);
 
+  const iconColorStyle = React.useCallback(() => {
+    if (isDisabled) return COLORS.gray300;
+    return COLORS.white2;
+  }, [isDisabled]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `React.useCallback` here seems unnecessary. The functions `iconBackgroundColorStyle` and `iconColorStyle` are not expensive computations, nor are they passed as props to child components that might cause unnecessary re-renders. Therefore, wrapping them in `useCallback` does not provide any performance benefits. Instead, you can directly assign the color values based on the conditions.

```diff
-  const iconBackgroundColorStyle = React.useCallback(() => {
-    if (isDisabled) return COLORS.gray200;
-    if (type === 'SIGNED') return COLORS.signed_primary;
-    return COLORS.anon_primary;
-  }, [isDisabled, type]);
-
-  const iconColorStyle = React.useCallback(() => {
-    if (isDisabled) return COLORS.gray300;
-    return COLORS.white2;
-  }, [isDisabled]);

+  const iconBackgroundColor = isDisabled ? COLORS.gray200 : (type === 'SIGNED' ? COLORS.signed_primary : COLORS.anon_primary);
+  const iconColor = isDisabled ? COLORS.gray300 : COLORS.white2;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->